pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -391,30 +391,6 @@ func adjustContainerResources(isAutopilot bool, declared, current *appsv1.Deploy
 	return resourcesChanged, nil
 }
 
-// processAllowlist add resources.limits to allowlist when renconciler-manager does not own this field
-func processAllowlist(isAutopilot bool, declared *appsv1.Deployment, allowList []string) []string {
```

Comment:
Why was this originally put in and why is it being removed now? Also, was this not covered by any unit tests?

Reply:
This is covered by unit test TestCompareDeploymentsToCreatePatchData. Since we change to SSA, we can only manage the fields that defined in our declared reconciler deployment. The previous assumption is that if we do not declare the resource limit fields, we should ignore the differences on this field, to avoid unnecessary PATCH. However this miss the case I mentioned in the description, when user reset the resource limits override, there is no resource limit fields defined in declared deployment, and we should not ignore the difference for this case. So this optimization logic should be removed.

Reply:
The reason I asked about unit tests is it seems like this was able to be removed without any testcases being changed/removed, but maybe I'm missing something. I see the `manualChangeResourceLimits` helper function was changed - is that related?

Autopilot manages resource `requests` but not `limits`, right?

Reply:
There is one test case "non-Autopilot, no override limits, manual change limits" expectation changed. This optimization used to be standard cluster only, Autopilot cluster forces the resource limits to be same with resource requests, so the processAllowlist function directly return if it is Autopilot cluster.

Reply:
Oh I didn't notice that flipped from `true` -> `false`, my mistake. What was the original motivation for allowing drift on these fields for non-autopilot clusters?

Reply:
We did not define the resource limits for some containers to make the resource burstable, so if we do not define a field, we are not able to manage it.